spring-cloud-app-broker-deployer-cloudfoundry/src/main/java/org/springframework/cloud/appbroker/deployer/cloudfoundry/CloudFoundryDeploymentProperties.java

```diff
@@ -114,7 +114,7 @@
 	/**
 	 * The buildpack to use for deploying the application.
 	 */
-	private String buildpack;
+	private String buildpack = "java_buildpack_offline";
```

Comment:
seems reasonable to me. @spikymonkey any concerns about this?

Reply:
What's the context around the need for the change? I would probably favour keeping things explicit rather than defaulting - we can't guarantee which buildpacks are going to be available in the target env, and I've certainly seen envs where `java_buildpack_offline` doesn't exist.

Reply:
CFJC now requires a build pack name in some API calls.

Reply:
The context is that buildpacks are not optional anymore, they have to be explicitly specified.
For example https://github.com/spring-cloud/spring-cloud-app-broker/pull/409/files#diff-6ed7594932ade9fa992e723b4f269c0b61cc4127a8dc40af23c9ab1168cc6e53R54
With that default we just make it work in most of the PCF environments, but it's true that `java_buildpack_offline` is not available in other non PCF environments.
We have to document it regardless of the default that `buildpack` is not optional anymore, regardless if we default it or not.

Reply:
Wait what? 😅

Reply:
That's inconvenient

Reply:
I can't really tell where the mandatory buildpack has come in from this, but it doesn't make much sense to me.  User should totally be able to throw some bits at the API and have the available buildpacks determine which if any can convert those bits into a package. Any chance we can use the buildpack returned from the build creation step in further API calls?

Reply:
To be more concise, the buildpacks are now mandatory when doing a push with manifest, which was not the case before and it's what we use to do most of the create operations. 
We don't have much information about buildpacks before that.

Reply:
Can it just be set to an empty array (i.e. autodetect buildpack)? 

![Screenshot 2021-02-18 at 5 30 15 pm](https://user-images.githubusercontent.com/4227599/108396691-06daa380-720f-11eb-9abd-c1ef0a22f3bd.png)

Reply:
Definitely much better! I will change it to that. Thanks!

Reply:
Updated it to an empty String to construct the manifest as an empty buildpacks array.